apps/ui/src/networks/offchain/helpers.ts

```diff
@@ -1,4 +1,52 @@
-import { Choice } from '@/types';
+import { StaticJsonRpcProvider } from '@ethersproject/providers';
+import { RpcProvider } from 'starknet';
+import { getProvider } from '@/helpers/provider';
+import { METADATA as STARKNET_METADATA } from '@/networks/starknet';
+import { ChainId, Choice, NetworkID } from '@/types';
+import { EDITOR_SNAPSHOT_OFFSET } from './constants';
+import { createProvider } from '../starknet/provider';
+
+const STARKNET_CHAIN_ID_TO_NETWORK_MAPPING: Record<ChainId, NetworkID> =
+  Object.fromEntries(
+    Object.entries(STARKNET_METADATA).map(([network, metadata]) => [
+      metadata.chainId as ChainId,
+      network as NetworkID
+    ])
+  );
+
+function getStarknetMetadata(chainId: string) {
+  return STARKNET_METADATA[
+    STARKNET_CHAIN_ID_TO_NETWORK_MAPPING[chainId as ChainId]
+  ];
+}
+
+export function isStarknetChainId(chainId: string): boolean {
+  return !!STARKNET_CHAIN_ID_TO_NETWORK_MAPPING[chainId as ChainId];
+}
+
+export async function getLatestBlockNumber(chainId: string): Promise<number> {
+  try {
+    let provider: StaticJsonRpcProvider | RpcProvider;
+
+    if (isStarknetChainId(chainId)) {
+      const starknetMetadata = getStarknetMetadata(chainId);
+      if (!starknetMetadata) {
+        throw new Error(`Error creating provider for chain ID: ${chainId}`);
+      }
+      provider = createProvider(starknetMetadata.rpcUrl);
+    } else {
+      provider = getProvider(Number(chainId));
+    }
+
+    const blockNumber = await provider.getBlockNumber();
+
+    return Math.max(0, blockNumber - EDITOR_SNAPSHOT_OFFSET);
```

Comment:
What is the idea behind EDITOR_SNAPSHOT_OFFSET?

For Ethereum 4 blocks is around 50 seconds, for Starknet mainnet it will be 2 minutes. Does it matter?